recipes/torchsde/meta.yaml

```
@@ -0,0 +1,52 @@
+{% set name = "torchsde" %}
+{% set version = "0.2.4" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/google-research/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 8a4f1f7dd3011bf6b3a1a8c97b9e8ab4093da79b863a83e022d00dd677cc0b7d
+
+build:
+  skip: True  # [win]
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - setuptools >=40.8.0
+  run:
+    - python
+    - blist
+    - boltons >=20.2.1
+    - numpy ~=1.19.0
+    - pytorch >=1.6.0
+    - trampoline >=0.1.2
+    - typing_extensions
+    - scipy ~=1.5.0
```

```suggestion
    - scipy 1.5
```
Are you really pinned exactly to scipy 1.5?

---

The challenge is that there are a few migrations underway, and 1.5 might make your environment unsolvable seeing as the latest package on conda-forge is 1.6.

---

As I understand it, `~=1.5.0` means a release [*compatible with*](https://www.python.org/dev/peps/pep-0440/#compatible-release) `1.5.0` which gets translated to `>=1.5.0,<1.6`

I believe `conda` translates `1.5` as `1.5.*` which is also equivalent to `>=1.5.0,<1.6` - so I think they're the same?

I tend to use the compatible release version as I think it's less ambiguous. i.e. I think it would be perfectly reasonable to expect `1.5` to be translated to `1.5.0` and a user just has to know that it actually means `1.5.*`. With the compatible release version the approximately equal to `~=` sign gives the reader a clue that it's not the same as exact equality.

---

If that's wrong our you'd still like the change I'm happy to commit it?

---

You can choose the notation you like. Are you sure you don't want scipy 1.6 as well?

---

It's what is specified upstream:
https://github.com/google-research/torchsde/blob/709968cc008f4bbfd6c63775974ac960b3746d4f/setup.py#L38

I tend to be conservative with going against upstream, particularly for compiled projects. That said, a lot of upstreams don't pin correctly (because it has never been an issue with `pip`).

---

Happy to loosen that to `>1.5,<1.7` if you think it will cause issues as is though?

---

[`5ef5ebc` (#86)](https://github.com/google-research/torchsde/pull/86/commits/5ef5ebca23fe20eab90094ffca186e371472e66f)

---

yeah, i figured that was your line of thought.